File: server/src/main/java/moe/yuuta/server/github/Release.java
package moe.yuuta.server.github;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * The release bean of GitHub API
 * NOTICE: It does not contain all attributes.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class Release {
    @JsonProperty("url")
    private String url;
    @JsonProperty("html_url")
    private String htmlUrl;
    @JsonProperty("id")
    private int id;
    @JsonProperty("name")
    private String name;
    @JsonProperty("body")
    private String body;
    @JsonProperty("tag_name")
    private String tagName;

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getHtmlUrl() {
        return htmlUrl;
    }

    public void setHtmlUrl(String htmlUrl) {
        this.htmlUrl = htmlUrl;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }

    public String getTagName() {
        return tagName;
    }

    public void setTagName(String tagName) {
        this.tagName = tagName;
    }

    @Override
    public String toString() {
        return "Release{" +
                "url='" + url + '\'' +
                ", htmlUrl='" + htmlUrl + '\'' +
                ", id=" + id +
                ", name='" + name + '\'' +
                ", body='" + body + '\'' +
                ", tagName='" + tagName + '\'' +
                '}';
    }
}

File: server/src/main/kotlin/moe/yuuta/server/github/Release.kt
package moe.yuuta.server.github

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

/**
 * The release bean of GitHub API
 * NOTICE: It does not contain all attributes.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
data class Release(
        @JsonProperty("url") var url: String = "",
        @JsonProperty("html_url") var htmlUrl: String = "",
        @JsonProperty("id") var id: Int = -1,
        @JsonProperty("name") var name: String = "",
        @JsonProperty("body") var body: String = "",
        @JsonProperty("tag_name") var tagName: String = ""
)